src/usr/share/netcheck/netcheck.py

```diff
@@ -127,296 +133,295 @@ def _dns_works(self, network):
-                'server %s. Attempting second query.' %
-                (query_names[0], network, nameservers[0]))
+                '_dns_works: First DNS query for %s failed on connection %s using '
+                'name server %s. Attempting second query.', query_names[0], connection_id,
+                nameservers[0])
             self.logger.trace(
-                '_dns_works: Attempting second DNS query for %s on interface %s using name '
-                'server %s.' % (query_names[1], network, nameservers[1]))
-            if (self._dns_query(network, nameservers[1], query_names[1])):
+                '_dns_works: Attempting second DNS query for %s on connection %s '
+                'using name server %s.', query_names[1], connection_id, nameservers[1])
+            if self._dns_query(connection_id, nameservers[1], query_names[1]):
                 dns_works = True
-                self.logger.trace('_dns_works: Second DNS query on %s successful.' %
-                                  network)
+                self.logger.trace(
+                    '_dns_works: Second DNS query on connection %s successful.',
+                    connection_id)
             else:
                 self.logger.debug(
-                    '_dns_works: Second DNS query for %s failed on interface %s using name '
-                    'server %s. Assuming network is down.' %
-                    (query_names[1], network, nameservers[1]))
+                    '_dns_works: Second DNS query for %s failed on connection %s using name '
+                    'server %s. Assuming connection is down.', query_names[1],
+                    connection_id, nameservers[1])
 
         return dns_works
 
-    def _connect_and_check_dns(self, network_name):
-        """Connects to a network and checks DNS availability if connection was successful.
-        Returns true on success, false on failure.
+    def _activate_connection_and_check_dns(self, connection_id):
+        """Activates a connection and checks DNS availability if the activation was
+        successful.
+
+        connection_id: The name of the connection as displayed in NetworkManager.
+        Returns True on successful DNS lookup, False otherwise.
         """
 
-        self.logger.trace('_connect_and_check_dns: Attempting to connect to and reach the '
-                          'Internet over %s.' % network_name)
+        self.logger.trace('_activate_connection_and_check_dns: Attempting to activate and '
+                          'reach the Internet over connection %s.', connection_id)
 
         overall_success = False
-        connection_successful = self.network_meta.connect(network_name)
-        if not(connection_successful):
-            self.logger.debug('_connect_and_check_dns: Could not connect to network %s.' %
-                              network_name)
+        activation_successful = self.network_helper.activate_connection(connection_id)
```

Comment:
We might want to put a try/except block here so we don't get stick on a single connection.